### ISSUE TITLE
AirflowNetwork:MultiZone:ExternalNode interaction with OutdoorAir:Node is confusing when using surface local environments

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#7656]
Issue overview
--------------
Input Output Reference for AirflowNetwork:MultiZone:ExternalNode needs to explain how it interacts with OutdoorAir:Node when there are any external environments present and maybe the code needs to change?

User was getting errors like this:
````   
** Severe  ** GetAirflowNetworkInput: When the choice of Height Selection for Local Wind Speed Calculation is OpeningHeight, the number of external surfaces defined in AirflowNetwork:MultiZone:Surface objects 
   **   ~~~   ** has to be equal to the number of AirflowNetwork:MultiZone:ExternalNode objects.
   **   ~~~   ** The entered number of external nodes is 22. The entered number of external surfaces is 18.
   **  Fatal  ** GetAirflowNetworkInput: Errors found getting inputs. Previous error(s) cause program termination.
````
Turns out that there is an undocumented (at least I couldn't find it anywhere in the I/O Ref) interaction between local environments and OutdoorAir:Node and AirflowNetwork:MultiZone:ExternalNode.

When using AirflowNetwork:Multizone with Local Wind Speed Calculation = OpeningHeight, you need an AirflowNetwork:MultiZone:ExternalNode for each exterior surface.

However, if there are any external environments (which includes external shading schedules), AirflowNetwork counts all OutdoorAir:Node objects as AirflowNetwork:MultiZone:ExternalNode objects. 

### Workaround ###
So, if you have any outdoor air node inlets that you don't want to use for airflownetwork, change them to OutdoorAir:NodeList. The ones that you do keep as OutdoorAir:Node should have all the relevant fields as they would for AirflowNetwork:MultiZone:ExternalNode, and then refer to these nodes in your AirflowNetwor...Surface objects and do not duplicate them in AirflowNetwork:MultiZone:ExternalNode.

### More Details ###
Relevant code is [here](https://github.com/NREL/EnergyPlus/blob/d0361cca92e8ea80e54da7e21067ce550ae5a391/src/EnergyPlus/AirflowNetworkBalanceManager.cc#L2344-L2398) and [error message here](https://github.com/NREL/EnergyPlus/blob/d0361cca92e8ea80e54da7e21067ce550ae5a391/src/EnergyPlus/AirflowNetworkBalanceManager.cc#L2878-L2889)

1. Seems this could be smarter, especially if the local environment objects are not used to specify an air node, or just detailed shading is used.
2. In any case, the impact on AirflowNetwork:MultiZone:ExternalNode needs to be explained in the I/O ref for AirflowNetwork:MultiZone:ExternalNode and in any relevant external environment objects.
3. The error message should be sensitive to the presence of external environments and add some language about that.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 9.2.0
 - Helpdesk ticket number 15326

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by lgu1234** [comment]
@xuanluo113 I am unable to figure out the relationship between local environments and OutdoorAir:Node and AirflowNetwork:MultiZone:ExternalNode. Could you take a look at it?

**Comment by xuanluo113** [comment]
@lgu1234 Based on the documentation, if local `OutdoorAir:Node` is declared, the surface node uses the local environment variables. In code implementation, they are extended after the `AirflowNetwork:MultiZone:ExternalNode` to the same array.

```
AirflowNetworkNumOfExtNode = GetNumObjectsFound("AirflowNetwork:MultiZone:ExternalNode";);
if ( AnyLocalEnvironmentsInModel ) {
AirflowNetworkNumOfOutAirNode = GetNumObjectsFound( "OutdoorAir:Node");
AirflowNetworkNumOfExtNode += AirflowNetworkNumOfOutAirNode;
```
Please refer to:
https://github.com/NREL/EnergyPlus/pull/6263#pullrequestreview-58372842

**Comment by lgu1234** [comment]
@xuanluo113 Let me understand. The test file inputs 3 OutdoorAir:Node object, so that 3 additional AirflowNetwork:MultiZone:ExternalNode are created. I am not clear which surface nodes are applied, existing surface nodes or new surfaces.

**Comment by xuanluo113** [comment]
@lgu1234 Should be existing surfaces.

**Comment by lgu1234** [comment]
@xuanluo113 Thanks for clarification. There are 2 test files for this issue at https://github.com/NREL/EnergyPlusDevSupport/tree/master/DefectFiles/7000s/7656. Could you take a look at them and let me know what is wrong?

In addition, do you have a test file for this feature and send it to me, so that I can understand more?

**Comment by xuanluo113** [comment]
@lgu1234 I'll take a look later this week. And this is the test file.
[AirflowNetwork_MultiZone_LocalNode.idf.txt](https://github.com/NREL/EnergyPlus/files/4282052/AirflowNetwork_MultiZone_LocalNode.idf.txt)

**Comment by lgu1234** [comment]
@xuanluo113 After looking at the example file, I think I understand your configuration. If an OutdoorAir:Node object is listed in the External Node Name field of the AirflowNetwork:MultiZone:Surface, the OutdoorAir:Node is treated as an AirflowNetwork External node. Otherwise, the other OutdoorAir:Node should be ignored. If my understanding is correct, I will take care of this issue.

Thanks.

**Comment by AgneseSalvati** [comment]
Hello, 
is there any update re this issue?
Thanks a lot for your help,
Agnese

**Comment by lgu1234** [comment]
@AgneseSalvati This issue has been fixed with PR at https://github.com/NREL/EnergyPlus/pull/7845. The Version 9.3 (release soon) will include the fix.

**Comment by rraustad** [comment]
Closed via #7845.

**Comment by AgneseSalvati** [comment]
Thanks!
